File: bit/bleak_numbers.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  int countBits(int n) {
    int ans = 0;
    while (n) {
      ++ans;
      n &= n - 1;
    }
    return ans;
  }

public:
  int is_bleak(int n) {
    // Code here.
    for (int i = 1; i < n; ++i) {
      if (n == i + countBits(i))
        return 0;
    }
    return 1;
  }
};

//{ Driver Code Starts.
int main() {
  int T;
  cin >> T;
  while (T--) {
    int n;
    cin >> n;
    Solution ob;
    int ans = ob.is_bleak(n);
    cout << ans << "\n";
  }
  return 0;
}

// } Driver Code Ends

File: bit/bleak_numbers.py
# User function Template for python3


class Solution:
    def is_bleak(self, n):
        # Code here
        def setbit(k):
            ans = 0
            while k > 0:
                ans += 1
                k &= k - 1
            return ans

        for i in range(1, 32):
            if setbit(n - i) == i:
                return 0
        return 1


# {
# Driver Code Starts
# Initial Template for Python 3

if __name__ == "__main__":
    T = int(input())
    for i in range(T):
        n = int(input())
        ob = Solution()
        ans = ob.is_bleak(n)
        print(ans)

# } Driver Code Ends
